backend/Techfesia2019/events/models.py

```diff
@@ -72,9 +70,113 @@ def save(self, *args, **kwargs):
 
 
 class SoloEvent(Event):
-    pass
+
+    @property
+    def event_type(self):
+        return 'single'
+
+    def current_participants(self):
+        return self.soloeventregistration_set.filter(is_confirmed=True)
+
+    def current_waiting_participants(self):
+        return self.soloeventregistration_set.filter(is_complete=True, is_confirmed=False)
+
+    def current_reserved_participants(self):
+        return self.soloeventregistration_set.filter(is_confirmed=True, is_reserved=True)
+
+    def current_waiting_reserved_participants(self):
+        return self.soloeventregistration_set.filter(is_complete=True, is_confirmed=False, is_reserved=True)
+
+    def refresh_participants(self):
+        total_seats = self.max_participants
+        total_reserved_seats = self.reserved_slots
+        # If reserved registrations are available
+        while self.current_waiting_reserved_participants().count() is not 0:
+            # If reserved seats are available
+            if self.current_reserved_participants().count() < total_reserved_seats:
```

Comment:
This seems to be a bit in-efficient as we are making db calls for each iteration to get the count.
Perhaps we can prefetch the number of vacant seats and then take that amount of people from waiting list and confirm them. That way we will require constant no of db calls

Reply:
yes we can slice the query list and make changes all together.
would do that.

Reply:
Is this one fixed?